cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/credential/CredentialProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.credential;
+
+import io.cdap.cdap.api.security.credential.CredentialIdentity;
+import io.cdap.cdap.api.security.credential.CredentialProvisioningException;
+import io.cdap.cdap.api.security.credential.ProvisionedCredential;
+import io.cdap.cdap.proto.credential.CredentialProfile;
+
+/**
+ * Defines an SPI for provisioning a credential.
+ */
+public interface CredentialProvider {
+
+  /**
+   * Returns the name of the credential provider.
+   *
+   * @return the name of the credential provider.
+   */
+  String getName();
+
+  /**
+   * Initializes the credential provider.
```

Comment:
should include more information about the contract. I would assume it gets called once before any other methods are called.

Reply:
Done, added note that the method is called exactly once. I'll also update the contract as we have more clarity on exactly what's expected of the initialization method.